jwt.js

```diff
@@ -175,7 +179,11 @@ function fastifyJwt (fastify, options, next) {
 
     steed.waterfall([
       function getSecret (callback) {
-        secretCallbackSign(reply.request, payload, callback)
+        const signResult = secretCallbackSign(reply.request, payload, callback)
```

Comment:
Checking `secretCallbackSign.length` would let us to avoid to provide the `callback` as parameter

I saw too many times `async (req, res, cb)` so the user will have `cb` as `undefined` and it will spot it during the developing

Reply:
Just so I understand what you mean, if the user provides `async (req, token, callback)` the length is 3 so we provide a callback, and yet the function returns a promise. Are you suggesting that when the function has 3 parameters we discard the return value even if it's a promise? I have a feeling that if we want to handle this rather than relying on the user to do something sensible we have other cases to cover. E.g. `(req, token)` which doesn't return a promise. I'm fairly sure this is a recurring pattern but honestly I haven't checked out how it's handled elsewhere, I just thought delegating to the user the responsibility of getting the signature right was the simplest reasonable thing to do

Reply:
Yes

- if the user's func has 3 arguments, we provide the callback and, even if the function returns a promise, the user must call the callback to go further -> so it can be understood while implementing that there is something wrong if the request timeout
- f the user's func has 2 arguments and the function doesn't return a promise, we can throw an error -> as previous this case can be spotted while developing

Instead and async function + callback could let the user call the `callback` and resolve the promise, so it could lead to a memory leak that can be spotted only in production after a bad out of memory 😓

Reply:
We typically don't cover this case in Fastify hooks. Unfortunately it's expensive to care for it.

Reply:
Folks if you're happy I'll leave it as-is. Handling this would look something similar to this, but I guess that'll open up additional debate:

```js
  function cbfy (fn, callback, ...args) {
    switch (fn.length) {
      // promise
      case 2: {
        const result = fn.apply(null, args)

        if (!result || typeof result.then !== 'function') {
          return callback(new Error('expected function to return a Promise'))
        }

        return result.then(result => callback(null, result), callback)
      }
      // callback
      case 3: {
        return fn.apply(null, args.concat([callback]))
      }
      default: {
        return callback(new Error('unsupported function signature'))
      }
    }
  }
```

Reply:
No problem, I was just suggesting 👍🏽 

I will think if that kind of snippet could become a single-liner utility 😄